messaging/src/main/java/org/axonframework/messaging/annotation/HandlerComparator.java

```diff
@@ -40,21 +43,20 @@ public final class HandlerComparator {
     private static final Comparator<MessageHandlingMember<?>> INSTANCE =
             Comparator.comparingInt((ToIntFunction<MessageHandlingMember<?>>) MessageHandlingMember::priority)
                       .reversed()
-                      .thenComparing(
-                              (Function<MessageHandlingMember<?>, Class<?>>) MessageHandlingMember::payloadType,
-                              HandlerComparator::compareHierarchy
-                      )
-                      .thenComparing(Comparator.comparingInt(
-                              (ToIntFunction<MessageHandlingMember<?>>) m -> m.unwrap(Executable.class)
-                                                                              .map(Executable::getParameterCount)
-                                                                              .orElse(1)
-                                     ).reversed()
-                      )
-                      .thenComparing(
-                              m -> m.unwrap(Executable.class)
-                                    .map(Executable::toGenericString)
-                                    .orElse(m.toString())
-                      );
+                      .thenComparing((memberOne, memberTwo) -> {
+                          Comparator<MessageHandlingMember<?>> handlerComparator =
+                                  Comparator.comparing(
+                                                    (Function<MessageHandlingMember<?>, Class<?>>) MessageHandlingMember::payloadType,
+                                                    HandlerComparator::compareHierarchy
+                                            )
+                                            .thenComparing(Comparator.comparingInt(HandlerComparator::parameterCount)
+                                                                     .reversed())
+                                            .thenComparing(HandlerComparator::executableName);
```

Comment:
Note that the name may not be 100% decisive in the comparison. Taking the entire signature into account does give that guarantee.

Reply:
Ah, I should've called the method `executableSignature` indeed. My bad.